### PR TITLE
Allow sharing disks between VMs

### PR DESCRIPTION
**Pull request by alicefr**
**What this PR does / why we need it**:
Add the shareable option for disks. This allows multiple VMs to share the same disk. 
The write synchronization is out-of-scope of this setup and it is the user's responsibility of using a proper configuration to prevent data corruption. The write synchronization needs to be managed either by the application itself or a cluster-aware filesystem.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Add shareable option to identify if the disk is shared with other VMs
```

### CODE REVIEW

**Comment by alicefr** [comment]
fyi @aglitke the functional tests are using the standard storage class and a data volume

File: tests/storage/storage.go [hunk 1]
```diff
@@ -1164,7 +1173,80 @@ var _ = SIGDescribe("Storage", func() {
...
+					return false
+				}, 120, 1).Should(BeTrue())
+			})
+		})
+		Context("write and read data from a shared disk", func() {
+			It("should successfully write and read data", func() {
+				vmi1 := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+				vmi2 := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+				diskName := "disk1"
+				tests.CreateBlockVolumePvAndPvc("1Gi")
```
Comment (maya-r) [inline tests/storage/storage.go:1221]: Wouldn't this suffer from the same problem described in #6739?

Comment (alicefr) [reply tests/storage/storage.go:1221]: definitely, need to change this

**Comment by alicefr** [comment]
I'd like to use the function for the tests defined in #6739

File: tests/storage/storage.go (hunk 1 above)
Comment (alicefr) [reply tests/storage/storage.go]: @maya-r I changed this to use the new function [CreateCephPVC](https://github.com/kubevirt/kubevirt/blob/e622b99f6cfe4cbd659ff408dffb411effa2664c/tests/utils.go#L5254) and it now uses a Ceph PVC

**Comment by maya-r** [comment]
/lgtm
might make sense to have a release note for it too

File: staging/src/kubevirt.io/api/core/v1/schema.go [hunk 2]
```diff
@@ -569,6 +569,9 @@ type Disk struct {
 	// If specified, the virtual disk will be presented with the given block sizes.
 	// +optional
 	BlockSize *BlockSize `json:"blockSize,omitempty"`
+	// If specified the disk is made sharable and multiple write from different VMs are permitted
+	// +optional
+	Shareable bool `json:"shareable,omitempty"`
```
Comment (mhenriks) [inline staging/src/kubevirt.io/api/core/v1/schema.go:574]: should this be `*bool` like `DedicatedIOThread`?

Comment (alicefr) [reply staging/src/kubevirt.io/api/core/v1/schema.go:574]: if I use a bool, it relies on the golang default and is false if not set. With the pointer, I need to check if it isn't `nil` or not and then set the value. I can do it!

**Comment by alicefr** [comment]
Certain new tests were timing out because the datavolume import was still in import phase. I increased the timeout of the function `RunVMIAndExpectLaunchWithDataVolume` from 90 to 500 as the other tests that uses this function

File: staging/src/kubevirt.io/api/core/v1/schema.go (hunk 2 above)
Comment (mhenriks) [reply staging/src/kubevirt.io/api/core/v1/schema.go]: yeah suggesting just for consistency sake

**Comment by alicefr** [comment]
There are some IO error occuring for the second VM in `"should fail starting the second VM when the disk is not shareable (default)"`, so the VM is pausing and I don't get the expected error message. I still investigating

**Comment by alicefr** [comment]
I removed the test `should fail to start the second VM when the disk is not shareable (default)`. It actually doesn't test the shareable functionality but more if the qemu locking works. On my system, it fails with `Failed to get \"write\" lock\nIs another process using the image` error message that I was checking in the test.On the CI it provokes some IO errors which make some sort of sense to me.

**Comment by alicefr** [comment]
I was wrongly assuming that vmis with the same RWO volume would be scheduled on the same node, but that's not the case. I added the affinity to the VMs that needs to share the same volume

**Comment by alicefr** [comment]
@maya-r @mhenriks do you have some time review this? I just added the affinity for the VM that needs to share the disk. In this way, we ensure that they are scheduled on the same node

File: pkg/virt-launcher/virtwrap/api/schema.go [hunk 3]
```diff
@@ -586,6 +588,7 @@ type Disk struct {
 	FilesystemOverhead *cdiv1.Percent `xml:"filesystemOverhead,omitempty"`
 	Capacity           *int64         `xml:"capacity,omitempty"`
 	ExpandDisksEnabled bool           `xml:"expandDisksEnabled,omitempty"`
+	Shareable          *Shareable     `xml:"shareable,omitempty"`
```
Comment (mhenriks) [inline pkg/virt-launcher/virtwrap/api/schema.go:591]: What is the purpose of this? I don't see it referenced anywhere.

Comment (alicefr) [reply pkg/virt-launcher/virtwrap/api/schema.go:591]: This is needed for define the field in the XML. It is used for example in https://github.com/alicefr/kubevirt/blob/shareable/pkg/virt-launcher/virtwrap/converter/converter.go#L191

**Pull request merged**